새로운 워크플로우 구조생성

yunseung-dable opened the pull request:
1. 입력값으로 디바이스를 받아 처리한다
2. 한개의 job 안에 세부 step들을  각각의 job으로 분리하여 배포의 진행상황을 관리하기 용이하다
3. 워크플로우를 재조합하여 사용하기에 구조적으로 용이

yunseung-dable committed:
kill process 웤플로우
권한변경 755

yunseung-dable committed:
start-process 추가
슬랙채널 조건문 수정
변수명 변경
테스트 웤플로우 생성

yunseung-dable committed:
get-vpn-ip 단게에서 입력값을 받아 처리할 수 있도록 변경

yunseung-dable committed:
step을 job 단위로 분리
name, id 하나의 scope로 묶기

yunseung-dable committed:
outputs 코드 수정
다음 스텝을 위한 inputs 항목 추가
output 으로 list를 받기 위해 verbose 옵션 추가

yunseung-dable committed:
scp 배포 까지 테스트

yunseung-dable committed:
매트릭스 적용 후 each-device 스크립트
1호기 ssh 접속을 위해 exp_continue 제거

yunseung-dable committed:
ssh 접속후 scp, install 작업 적용
edge-setup check-file-existence 한글로 인해 일단 보류

yunseung-dable committed:
input : version 추가
version 파일에 버전이 저장되지않는 이슈 -> 논의 필요

yunseung-dable committed:
배포구조 완성

yunseung-dable committed:
병렬실행,
다른 호스트에서 실패해도 지속하도록
max-parallel 제거

yunseung-dable committed:
기존 작업을 step으로 나눠서 재활용성 가독성 증가
캡션 정리
check 코드 복구-> vv-edge-setup 명령어어와 버전이 달라 적용불가

yunseung-dable committed:
시작과 마지막에 슬랙보이로 알람 보내기

hyeseung-of-dable committed:
thread 댓글 메세지 수정

hyeseung-of-dable committed:
version update

hyeseung-of-dable committed:
Merge branch 'hotfix-slack'

yunseung-dable committed:
버전업
인자전달이 잘못되었을때 help 보여주도록 추가

jingu-lee commented on .github/workflows/deploy-to-each-device.yml:
```diff
@@ -0,0 +1,135 @@
+        description: exact name of process, using start and kill
+        required: true
+        type: string
+    secrets:
+      password:
+        description: edge device password
+        required: true
+      otp:
+        description: edge device otp
+        required: true
```
이건 얘기해볼 부분이긴 한데, self-hosted-runner로 배포를 하기로 했다면 굳이 각 레포에서 이 secrets를 관리하지 않아도 될 것 같습니다

yunseung-dable replied:
마자용, 그런데 secret을 organization 단위에서 설정하려다가 지금 흐지부지 된거 아닌가요?

jingu-lee replied:
organization 단위의 secrets는 self-hosted runner를 안 쓸 때 필요하고, self-hosted runner를 쓰면 내부적으로 secrets를 가지고 있을 수 있습니다.

hyeseung-of-dable replied:
self-hosted runner에서 다른 장비들의 pwd를 관리하는게 보안상 좋지않아, 여전히 secrets를 사용하고있습니다

jingu-lee commented on deploy-file.sh:
```diff
@@ -0,0 +1,56 @@
+		HOST=$2
+		shift 2
+		;;
+  --user)
```
indent가 다른 라인하고 다릅니다

yunseung-dable replied:
이게 ide 상으론 똑같은데 github에 올리면 다르게 나타날때가 종종 있더라구요!
PyCharm 만 그런줄 알았는데 vscode도 그렇다는걸 찾아서 나중에 비슷한 경험하실때 참고하시면 좋을것 같습니다
https://github.com/github/VisualStudio/issues/1524

jingu-lee commented on deploy-file.sh:
```diff
@@ -0,0 +1,56 @@
+LONG=host:,user:,password:,otp:,zip-file-name:
+OPTS=$(getopt -o '' -a --longoptions $LONG  -- "$@")
+
+[ $? -eq 0 ] || {
+    echo "인자전달이 잘못되었습니다. "
+    exit 1
+}
```
저희 대부분의 스크립트가 이렇게 작성되어 있기는 한데, 사실 인자 전달이 잘못되었으면 help를 띄워주는 것이 보편적인 듯 합니다. actions에서 다룬다고 할 지라도 인자가 하나가 아니기 때문에 ;잘못되었다' 보다는 현재 전달된 인자를 보여주는 방식도 좋아 보입니다.

yunseung-dable replied:
진구님 친절한 개발자시군요

jingu-lee commented on deploy-to-each-device.sh:
```diff
@@ -0,0 +1,112 @@
+		HOST=$2
+		shift 2
+		;;
+  --user)
```
이곳도 인덴트!

jingu-lee reviewed (commented):
작성 해놓은 내용들 중 윤승님 판단하기에 수정이 필요한 부분만 수정해서 다시 리퀘스트 주세요

jingu-lee reviewed (approved):
저는 큰 문제가 없어 보이긴 하는데, 제가 CD 구조를 몰라서 혜승님이나 다른 분들도 리뷰하신 후 머지하는 게 좋아보입니다.

hyeseung-of-dable committed:
여러 파일을 zip에서 delete할 수 있도록 수정. resolve #74

hyeseung-of-dable committed:
version update. v1.1.6

hyeseung-of-dable committed:
Merge branch 'hotfix-zip-delete'

hyeseung-of-dable reviewed (changes requested):
1. deploy-to-each-device.sh 는 사용하지않는 것 같은데, 만드신 이유가 있을까요?! 
2. readme 작성 요청드립니다 🙇

yunseung-dable committed:
PR 리뷰 반영
1. 안쓰는 주석코드제거
2. 버전업
3. 슬랙보이 syntax 에러, 메세지수정
+ get_vpn_ip argument action 변경
4.deploy, install시 체크프로세스 포함
5.버전파일 체크 추가
6.리드미 추가, 팀정보 커버링

yunseung-dable commented:
리뷰코드 반영 하려고 보니 몇가지 추가 되었으면 하는 기능이 생각나서 공유드립니다.
1. vpn.ini 파일 대신 입력으로 받으며 live 장비 전체배포 기능이 사라져서 이번 merge 이후에 추가하면 좋을것 같습니다! 
2. github actions outputs 기능을 활용하여 version check, install-check 부분을 간소화할 수 있을것 같습니다!

hyeseung-of-dable commented on README.md:
```diff
@@ -189,3 +193,53 @@ jobs:
+        uses: teamdable/vv-deploy-actions/.github/workflows/parallel_deployment.yml@main
+        with:
+          deploy-target-device: ${{ github.event.inputs.deploy-target-devices }}
+          user: 'vv-team'
+          code-name: 'resource'
+          deploy-branch: ${GITHUB_REF##*/}
+          parent-dir: '/home/vv-team/monitoring'
+          version-file-name: '_version.py'
+          exclude-files-from-zip: 'bin/deploy/vpn-config.ini'
+          slack-channel: 'dev-vv-lounge'
+          process-name: 'resource-monitoring'
```
퍼블릭이라 계정명등을 올리지않고있습니다~~
reset 했다가 커밋다시올려주세요!

기존 예시)
```
uses: teamdable/vv-deploy-actions/.github/workflows/deploy-to-edge-devices.yml@main
with:
    user: 'ubuntu'
!   code-name: 'inference'
!   deploy-branch: ${GITHUB_REF##*/} 
!   parent-dir: '/home/ubuntu/'
!   version-file-name: '_version.py'
-   exclude-files-from-zip: 'bin/deploy/vpn-config.ini'
+   
!   slack-channel: 'general'
```

hyeseung-of-dable reviewed (changes requested):
@yunseung-dable 

1. https://github.com/teamdable/vv-deploy-actions/issues/22 요 이슈를 어느정도해결해주셨더라구요 ㅎㅎ live 포함하는 부분은, 이슈 댓글처럼 get-vpn-ip쪽에서 수정 + 배포할 때, 레포에서 사용자가 live포함할지안할지 t/f입력하도록 수정할 예정입니다!
2. 좋씁니다 !! 그러면 expect때문에 불안한것도 어느정도 해소되겠네요!

hyeseung-of-dable commented:
수고하셨습니다 ~~🚀🚀🚀
